lib/child_process.js

```diff
@@ -183,23 +183,22 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
-    let resolve;
-    let reject;
-    const promise = new Promise((res, rej) => {
-      resolve = res;
-      reject = rej;
+  return (...file) => {
+    let child;
+
+    const promise = new Promise((resolve, reject) => {
+      child = orig(...file, (err, stdout, stderr) => {
+        if (err !== null) {
+          err.stdout = stdout;
+          err.stderr = stderr;
+          reject(err);
+        } else {
+          resolve({ stdout, stderr });
+        }
+      });
     });
 
-    promise.child = orig(...args, (err, stdout, stderr) => {
-      if (err !== null) {
-        err.stdout = stdout;
-        err.stderr = stderr;
-        reject(err);
-      } else {
-        resolve({ stdout, stderr });
-      }
-    });
+    promise.child = child;
```

Comment:
This change seems unrelated – and IMHO the current version is better than the one proposed here. Can you revert this please?

Reply:
Did you refer to the promise instantiation and execution of the `orig` function?

Reply:
My understanding – and crrect me if I'm wrong – is that is just a refactor of the `customPromiseExecFunction` function, a refactor that doesn't change the code behaviour (and thus is not necessary for this PR). If it's the case, I'd rather have you revert it to the original value in this PR, and optionally open another PR for that refactor if you think it improves the code.

Reply:
Got it, has totally sense. I'll revert it

Reply:
Done in 2305734
Please have a look :)